app/models/mhv_account.rb

```diff
@@ -56,12 +56,20 @@ class MhvAccount < ActiveRecord::Base
   end
 
   def create_and_upgrade!
-    create_mhv_account! unless preexisting_account?
-    upgrade_mhv_account!
+    if identity_proof_valid? && va_patient?
+      create_mhv_account! unless preexisting_account?
+      upgrade_mhv_account!
+    end
   end
 
+  # if you already have an MHV account you're eligible to use it on vets.gov
+  # but if you don't have one, we will only create / upgrade if you're a va patient.
   def eligible?
-    va_patient?
+    va_patient? || preexisting_account?
+  end
+
+  def identity_proof_valid?
```

Comment:
This logic should definitely not live in `mhv_account`

Reply:
I agree, I mostly put it in here to call attention to it. I don't think it hurts to have this check in here for logic readability, but yes, it probably ought to be a method on user.rb perhaps to replace any validations we remove.